libraries/persistent_store/src/lib.rs

```diff
@@ -12,299 +12,297 @@
+//! -   Given a key and a value, [`StoreUpdate::Insert`] updates the store such that the value is
 //!     associated with the key. The values for other keys are left unchanged.
-//! -   Given a key, `Remove` updates the store such that no value is associated with
-//!     the key. The values for other keys are left unchanged. Additionally, if there
-//!     was a value associated with the key, the value is wiped from the storage
-//!     (all its bits are set to 0).
+//! -   Given a key, [`StoreUpdate::Remove`] updates the store such that no value is associated with
+//!     the key. The values for other keys are left unchanged. Additionally, if there was a value
+//!     associated with the key, the value is wiped from the storage (all its bits are set to 0).
 //!
 //! The store provides the following _read-only operations_:
-//! -   `Iter` iterates through the store returning all entries exactly once. The
-//!     iteration order is not specified but stable between mutable operations.
-//! -   `Capacity` returns how many words can be stored before the store is full.
-//! -   `Lifetime` returns how many words can be written before the storage lifetime
-//!     is consumed.
+//! -   [`Store::iter`] iterates through the store returning all entries exactly once. The iteration
+//!     order is not specified but stable between mutable operations.
+//! -   [`Store::capacity`] returns how many words can be stored before the store is full.
+//! -   [`Store::lifetime`] returns how many words can be written before the storage lifetime is
+//!     consumed.
 //!
 //! The store provides the following _mutable operations_:
-//! -   Given a set of independent updates, `Transaction` applies the sequence of
-//!     updates.
-//! -   Given a threshold, `Clear` removes all entries with a key greater or equal
-//!     to the threshold.
-//! -   Given a length in words, `Prepare` makes one step of compaction unless that
-//!     many words can be written without compaction. This operation has no effect
-//!     on the store but may still mutate its storage. In particular, the store has
-//!     the same capacity but a possibly reduced lifetime.
-//!
-//! A mutable operation is _atomic_ if, when power is lost during the operation, the
-//! store is either updated (as if the operation succeeded) or left unchanged (as if
-//! the operation did not occur). If the store is left unchanged, lifetime may still
-//! be consumed.
+//! -   Given a set of independent updates, [`Store::transaction`] applies the sequence of updates.
+//! -   Given a threshold, [`Store::clear`] removes all entries with a key greater or equal to the
+//!     threshold.
+//! -   Given a length in words, [`Store::prepare`] makes one step of compaction unless that many
+//!     words can be written without compaction. This operation has no effect on the store but may
+//!     still mutate its storage. In particular, the store has the same capacity but a possibly
+//!     reduced lifetime.
```

Comment:
Out of curiosity: This looks like the line length limit was something 80 < x < 100 before, and is now 100? #176 suggests that we might want to unify this sometime.

Reply:
Yes indeed, I don't know why it was 80 before. The Rust default is 100 and this is what our `rustfmt.toml` file thus implicitly specifies. I think it was 80 because I first wrote this doc in a markdown file where the line length was 80, then copied it here without rewrapping everything.